### ISSUE TITLE
Component float-fieldtype does not exist

### ISSUE THREAD

**Issue by felixstoessel** [statamic/cms#3145]
## Bug Description
A weird bug appears and I have the theory that while updating Statamic, it doesn't really update everything. I get `Component float-fieldtype does not exist.` although I am on Statamic 3.0.38 (see screenshot). I also see the `float` fieldtype inside the backend in the blueprints. But somehow the CP JavaScript file has some issues.

There also was a bug a while back that inserted whitespaces in Bard above a newly inserted set. This was seemingly fixed, but the bug still appears here. I slowly start to wonder if this not 100% of Statamic get's updated when I update Statamic via `composer update statamic/cms --with-dependencies`...

## Extra Detail
![weird](https://user-images.githubusercontent.com/13189565/105505275-0d860180-5cc9-11eb-9700-9faee8a91e24.png)

## Environment
Statamic 3.0.38 Solo
Laravel 8.21.0
PHP 7.4.10

**Comment by damcclean** [comment]
Is it still broken if you run `php artisan statamic:install` and try again?

**Comment by felixstoessel** [comment]
I run this and I get:

```
Addon manifest generated successfully.
Copied Directory [/vendor/statamic/cms/resources/users] To [/resources/users]
Publishing complete.
Can't locate path: </home/vagrant/foobar-project/vendor/statamic/cms/src/Providers/../../resources/dist>
Publishing complete.
Compiled views cleared!
Application cache cleared!
```

The `Can't locate path` line in red, the color of success :)

**Comment by damcclean** [comment]
Ah, that'll be the issue. Can you share your `composer.json` file?

**Comment by felixstoessel** [comment]
```
{
    "name": "statamic/statamic",
    "description": "Statamic",
    "keywords": ["statamic", "cms", "flat file", "laravel"],
    "type": "project",
    "require": {
        "php": "^7.3 || ^8.0",
        "aryehraber/statamic-color-extractor": "^1.0",
        "fideloper/proxy": "^4.2",
        "fruitcake/laravel-cors": "^2.0",
        "guzzlehttp/guzzle": "^7.0.1",
        "laravel/framework": "^8.0",
        "laravel/tinker": "^2.0",
        "statamic/cms": "^3.0",
        "felixstoessel/backend": "*"
    },
    "require-dev": {
        "barryvdh/laravel-debugbar": "^3.5",
        "facade/ignition": "^2.3.6",
        "fakerphp/faker": "^1.9.1",
        "mockery/mockery": "^1.3.1",
        "nunomaduro/collision": "^5.0",
        "phpunit/phpunit": "^9.3"
    },
    "repositories": [{
      "type": "path",
      "url": "felixstoessel/backend"
    }],
    "config": {
        "optimize-autoloader": true,
        "preferred-install": "dist",
        "sort-packages": true
    },
    "extra": {
        "laravel": {
            "dont-discover": []
        }
    },
    "autoload": {
        "psr-4": {
            "App\\": "app/",
            "Database\\Factories\\": "database/factories/",
            "Database\\Seeders\\": "database/seeders/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
    "minimum-stability": "dev",
    "prefer-stable": true,
    "scripts": {
        "post-autoload-dump": [
            "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
            "@php artisan package:discover --ansi",
            "@php artisan statamic:install --ansi"
        ],
        "post-root-package-install": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ],
        "post-create-project-cmd": [
            "@php artisan key:generate --ansi"
        ]
    }
}
```

**Comment by damcclean** [comment]
Looks like it should work but let's try something...

If you change the version of `statamic/cms` from `^3.0` to `3.0.*` instead and run `composer update`, does it work?

**Comment by felixstoessel** [comment]
Okay, after long hours of debugging, I think I got the issue. There is an issue with macOS Catalina, the file system and VirtualBox. https://github.com/hashicorp/vagrant/issues/10961

When doing an `composer update`, it just didn't install everything. Why? No clue, but after adding `/sbin/nfsd` and VirtualBox to 'Full Disk Access' in 'Security & Privacy', everything seemed to update fully. Some older already fixed bugs are now implemented because some files weren't updated and/or failed silently. Technology in 2021 everybody!

I hope this is fixed now. Wish me luck everyone!
